File: Scrabble-v2/web/src/app/invitation.service.ts
import { Injectable } from '@angular/core';
import { MatDialog } from '@angular/material/dialog';
import { Room } from './classes/room';
import { InvitationPopupComponent } from './invitation-popup/invitation-popup.component';
import { Invitation } from './invite-friends/invite-friends.component';
import { SocketClientService } from './services/socket-client.service';

@Injectable({
    providedIn: 'root',
})
export class InvitationService {
    isInRoom: boolean;
    room: Room;
    constructor(room: Room, private socketService: SocketClientService, private dialog: MatDialog) {
        this.initializeRoom(room);
    }
    initializeRoom(room: Room) {
        // set up
        this.room = room;
        //
    }
    listenInvitation() {
        this.socketService.on('invite-friend', (invitation: Invitation) => {
            if (this.isInRoom) return;
            this.initializeRoom(invitation.room);
            this.dialog.open(InvitationPopupComponent, {
                data: invitation,
            });
        });
    }
}

File: Scrabble-v2/mobile/lib/services/invitation_service.dart
import 'dart:developer';

import 'package:flutter/material.dart';
import 'package:poly_scrabble/models/room.dart';
import 'package:poly_scrabble/models/user.dart';
import 'package:poly_scrabble/services/socket_service.dart';

class InvitationService extends ChangeNotifier {
  final Room room;
  final SocketService socketService;
  final UserModel userModel;
  InvitationService(this.room, this.socketService, this.userModel) {}
  initializeRoom(Room room) {
    //set up
    room = room;
    //
  }

  listenInvitation() {
    log("listenInvitation");
    socketService.on('invite-friend',
        (invitation) => {initializeRoom(room), log("got the invitation")});
  }
}
